tesseract-server/src/logic_layer/cache.rs

```diff
@@ -5,12 +5,101 @@ use log::info;
 
```

Comment:
Everything here appears to be good, but it is a pretty big diff :) I think that if there's errors, they will probably need to be shaken out (are there any places to test?), and otherwise the structure feels alright.

Reply:
Can you elaborate on your comment about shaking out errors?

Reply:
oh, it's just about logic errors that might be hard to catch until the user makes a particular request. Like duplicate drilldowns, etc.